Framework/Kernel/src/TimeROI.cpp

```diff
@@ -71,6 +71,14 @@ TimeROI::TimeROI(const Types::Core::DateAndTime &startTime, const Types::Core::D
 
 TimeROI::TimeROI(const Kernel::TimeSeriesProperty<bool> *filter) { this->replaceROI(filter); }
 
+TimeROI::TimeROI(const SplittingIntervalVec &splitter, const bool skipEmpty) {
```

Comment:
I believe this is conceptually wrong because it ignores the destination-workspace indexes of the splitter's element. For instance, the following splitter:
```
index      1        -1    3     -1
time    ---[---------)----[------)-------> time
```
ignores indexes `1` and `3` and creates a TimeROI with the two intervals. I believe this is a "time-ticking" bomb.
I think we should have instead 
```
TimeROI::TimeROI(const std::vector<TimeInterval> &intervals, const bool skipEmpty) ;
```
Places in the code where a `SplittingIntervalVec` is needless instantiated should be changed to `std::vector<TimeInterval>`. If this is too much work, then one could instantiate a ``std::vector<TimeInterval>` off the `SplittingIntervalVec` and then fed to the `TimeROI` constructor.

Reply:
This is converting a `SplittingIntervalVec` into a `TimeROI` rather than the other way around. Since `TimeROI` is only use/ignore, there isn't any place to put the indices. The constructor was added to make transitioning more of the existing code easier (read: less changes) so the old methods that ignore the index could be removed. I'd agree that the constructor you suggest would be more conceptually correct, but it is [YAGNI](https://www.martinfowler.com/bliki/Yagni.html) for now since no existing code creates a `std::vector<TimeInterval>` and this PR modified code that would (that I found) to directly using a `TimeROI` from the start.